### ISSUE TITLE
Question How to use this on non-flakes system ?

### ISSUE THREAD

**Issue by 573** [tweag/gomod2nix#16]
**Is your feature request related to a problem? Please describe.**
I want to build https://github.com/mickael-menu/zk via gomod2nix.
First problem was how to get the gomod2nix binary.
When I had a nix-shell running with gomod2nix finally I wanted to utilize the generated gomod2nix.toml in my checkout of zk.
Latter which fails as I am not able to utilize the overlay correctly, in terms of as the README here says.

**Describe the solution you'd like**
It would like to bind the overlay here in my zk checkout's default.nix such that nix-build can be satisfied.

**Describe alternatives you've considered**
This is the default.nix I'm using (`nix-build` it):
```nix
let
  pkgs = import <nixpkgs> {
    overlays = [
      (self: super: {
        buildGoApplication = pkgs.callPackage (builtins.fetchTarball {
          url =  https://github.com/tweag/gomod2nix/archive/67f22dd738d092c6ba88e420350ada0ed4992ae8.tar.gz;
          sha256 = "0a9rhw6djgcrky5nzly3srsl8095q42f2nyzxyisp9fh9fg70nih";
        }) {};
      })
    ];
  };
in pkgs.buildGoApplication {
  pname = "zk";
  version = "master";
  src = ./.;
  modules = ./gomod2nix.toml;
}
```

**Additional context**
Regarding the gomod2nix binary I started a nix-shell of https://github.com/GTrunSec/photoprism2nix and added `buildInputs = [ pkgs.gomod2nix ];` to `devShell` there in the flake.nix file. In the nix-shell started I can use `gomod2nix`.

**Comment by 573** [comment]
Tl;dr was missing the `/builder` suffix in my default.nix, correct version here:

```nix
let
  pkgs = import <nixpkgs> {
    overlays = [
      (self: super: {
        buildGoApplication = pkgs.callPackage ("${builtins.fetchTarball {
          url =  https://github.com/tweag/gomod2nix/archive/67f22dd738d092c6ba88e420350ada0ed4992ae8.tar.gz;
          sha256 = "0a9rhw6djgcrky5nzly3srsl8095q42f2nyzxyisp9fh9fg70nih";
        }}/builder") {};
      })
    ];
  };
in pkgs.buildGoApplication {
  pname = "zk";
  version = "master";
  src = ./.;
  modules = ./gomod2nix.toml;
  doCheck = false;
  CGO_ENABLED = "1";
}
```

See #17 for a follow-up regarding the `gomod2nix` binary.
